nvflare/app_common/workflows/cyclic_ctl.py

```diff
@@ -264,11 +265,11 @@ def restore(self, state_data: dict, fl_ctx: FLContext):
         finally:
             pass
 
-    def handle_dead_job(self, client_name: str, fl_ctx: FLContext):
-        super().handle_dead_job(client_name, fl_ctx)
-
-        new_client_list = []
-        for client in self._participating_clients:
-            if client_name != client.name:
-                new_client_list.append(client)
-        self._participating_clients = new_client_list
+    def handle_event(self, event_type, fl_ctx):
+        if event_type == EventType.JOB_DEAD:
+            client_name = fl_ctx.get_prop(FLContextKey.DEAD_JOB_CLIENT_NAME)
+            new_client_list = []
+            for client in self._participating_clients:
+                if client_name != client.name:
+                    new_client_list.append(client)
+            self._participating_clients = new_client_list
```

Comment:
We might also need to add "super().handle_event(event_type, fl_ctx)" in the end so whatever logic the parent component has will be invoked.

Reply:
We should try to avoid this as much as possible because workflow writers won't remember to do so.